core/src/main/scala/kafka/controller/TopicDeletionManager.scala

```diff
@@ -17,73 +17,55 @@
 package kafka.controller
 
 
+import kafka.common.TopicAndPartition
 import kafka.server.ConfigType
-import org.apache.kafka.common.protocol.Errors
-import org.apache.kafka.common.requests.{AbstractResponse, StopReplicaResponse}
-
-import collection.mutable
-import collection.JavaConverters._
-import kafka.utils.{Logging, ShutdownableThread}
-import kafka.utils.CoreUtils._
+import kafka.utils.Logging
 import kafka.utils.ZkUtils._
 
-import collection.Set
-import kafka.common.TopicAndPartition
-import java.util.concurrent.locks.ReentrantLock
-import java.util.concurrent.atomic.AtomicBoolean
+import scala.collection.{Set, mutable}
 
 /**
  * This manages the state machine for topic deletion.
  * 1. TopicCommand issues topic deletion by creating a new admin path /admin/delete_topics/<topic>
  * 2. The controller listens for child changes on /admin/delete_topic and starts topic deletion for the respective topics
- * 3. The controller has a background thread that handles topic deletion. The purpose of having this background thread
- *    is to accommodate the TTL feature, when we have it. This thread is signaled whenever deletion for a topic needs to
- *    be started or resumed. Currently, a topic's deletion can be started only by the onPartitionDeletion callback on the
- *    controller. In the future, it can be triggered based on the configured TTL for the topic. A topic will be ineligible
+ * 3. The controller's ControllerEventThread handles topic deletion. A topic will be ineligible
  *    for deletion in the following scenarios -
- *    3.1 broker hosting one of the replicas for that topic goes down
- *    3.2 partition reassignment for partitions of that topic is in progress
- *    3.3 preferred replica election for partitions of that topic is in progress
- *    (though this is not strictly required since it holds the controller lock for the entire duration from start to end)
+  *   3.1 broker hosting one of the replicas for that topic goes down
+  *   3.2 partition reassignment for partitions of that topic is in progress
+  *   3.3 preferred replica election for partitions of that topic is in progress
```

Comment:
3.3 is no longer true since a preferred replica election can't be in process when a topic deletion event is being handled. Ditto in 4.2.